frontend/src/components/ViewTrips/index.js

```diff
@@ -23,10 +28,37 @@ function getUserEmail() {
  * ViewTrips component that defines the page where a user can view and manage
  * their current trips.
  */
-const ViewTrips = () => {
-  return (
-    <TripsContainer db={db} userEmail={getUserEmail()} />
-  );
-};
+class ViewTrips extends React.Component {
+  /** @inheritdoc */
+  constructor() {
+    super();
+    this.state = { showModal: false };
+  }
+
+  /** Property that sets `showModal` to true --> displays add trip page. */
+  showAddTripModal = () => { this.setState({ showModal: true }); }
+
+  /** Property that sets `showModal` to false --> hides add trip page. */
+  hideAddTripModal = () => { this.setState({ showModal: false }); }
```

Comment:
I should probably use this syntax instead of function()

Reply:
Yeah I initially tried having this as a component member function and stuff wasn't working so I tried this an it seems to work nicely.